### DISCUSSION TITLE
The generated query string

### DISCUSSION THREAD

**Discussion by kazuser** [SchneiderInfosystems/FB4D#106] [Q&A] [answered]
Is there any way to see the generated query string (for "GET", "PUT", etc.) before it will be send to Firebase?

**Comment by SchneiderInfosystems** [comment, marked as answer]
All requests for Get/Put etc are sent to the Firebase service by the methods TFirebaseRequest.InternalSendRequest (asynchronous) or TFirebaseRequest.InternalSendRequestSynchronous. When using the debug version you can see a log entry for each request by using the DebugView tool from SysInternals.  (e.g. In Line 243/306/394  of FB4DRequest). Some API Calls transfer JSON data in the body. This is not logged. For inspecting these values please use the debugger.

**Reply by kazuser** [reply]
Maybe it makes sense to add a separate tab with the memo (in \Samples\Intro) and display the generated lines directly there? It would be clearer and more convenient. What do you think? 🤔

![изображение](https://user-images.githubusercontent.com/3335273/123955187-278be480-d9cb-11eb-8c6e-d6a6f964f91b.png)

**Reply by SchneiderInfosystems** [reply]
For me this log is only a debug helper tool. All information what you normally need is integrated into the class interfaces of the services.

Of course, I could implement such an internal log system and demonstrate it in the demo app. I'm unsure about the benefit of this work.

**Reply by kazuser** [reply]
Ok it's up to you of course.

Thanks anyway. FB4D is awesome! 👍

**Reply by SchneiderInfosystems** [reply]
It's like always: you should implement only the right and important things and focus on features that take products further. Sure, if a log function helps to get the library up and running faster, it can also have a benefit. On the other hand, it also makes the interface more complex. I'm a bit split, but would a solution that is presented as a pull request, certainly examine closely and build in if necessary. 
Update: After I wrote this comment I saw that there is exactly such a PR with an easy solution from Martin. I have merged that immediately. Please Check the newest function and add your logger to the TFirebaseHelpers.OnLog. This log is still limited to the Debug Version. I will think about this point too.
